extensions/http(s)/test/ace/http/request_test.exs

```diff
@@ -134,4 +134,21 @@ defmodule Ace.HTTP.RequestTest do
     assert_receive %{host: "www.raxx.com", path: [], body: nil, peer: ^client}
     assert_receive %{host: "www.raxx.com", path: [], body: nil, peer: ^client}
   end
+
+  test "can send response even when request headers are sent, content-length is non-zero but entire body is not sent", %{port: port} do
+    content = "Hello, World!\r\n"
+    {first, second} = Enum.split(content |> String.split(""), 7)
+    head = """
+    GET / HTTP/1.1
+    Host: www.raxx.com
+    Content-Length: #{:erlang.iolist_size(content)}
+
+    """
+    {:ok, socket} = :gen_tcp.connect({127,0,0,1}, port, [:binary])
+    :gen_tcp.send(socket, head <> Enum.join(first))
+    :timer.sleep(10)
+    assert_receive %{host: "www.raxx.com", path: [], body: ^content}
```

Comment:
Yep this is pretty much what is needed. In the "actual test" you might want to take a look at what is going on here https://github.com/CrowdHailer/Ace/blob/master/test/ace/http2_test.exs#L94